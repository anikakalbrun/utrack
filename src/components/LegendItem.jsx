import React from 'react';
import {string} from 'prop-types';

LegendItem.propTypes = {
    color: string.isRequired,
    label: string.isRequired
};

export default function LegendItem({color, label}) {
    return (
        <div className='legend-item'>
            <div style={{backgroundColor: color}} className='legend-box'/>
            <div className='legend-text'>{label}</div>
        </div>
    );
}