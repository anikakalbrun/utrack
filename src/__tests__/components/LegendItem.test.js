import React from 'react';
import LegendItem from '../../components/LegendItem';

let wrapper;
const props = {
    color: 'red',
    label: 'Cumulative'
};

describe('LegendItem', () => {

    it('renders correctly', () => {
        wrapper = renderer.create(<LegendItem {...props}/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

});
