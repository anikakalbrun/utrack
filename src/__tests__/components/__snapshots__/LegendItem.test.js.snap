// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegendItem renders correctly 1`] = `
<div
  className="legend-item"
>
  <div
    className="legend-box"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  />
  <div
    className="legend-text"
  >
    Cumulative
  </div>
</div>
`;
